libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/MetadataUtils.java

```diff
@@ -0,0 +1,175 @@
+
+        Pattern isIntRegExp = Pattern.compile("^\\d+$");
+
+        String width = attrs.getAttribute("width", "");
+        if (!isIntRegExp.matcher(width).matches() || NumberUtils.toInt(width) == 0) {
+            addMetadataProperty(metadata, "width", naturalWidth);
+        } else {
+            addMetadataProperty(metadata, "width", width);
+        }
+
+        String height = attrs.getAttribute("height", "");
+        if (!isIntRegExp.matcher(height).matches() || NumberUtils.toInt(height) == 0) {
+            addMetadataProperty(metadata, "height", naturalHeight);
+        } else {
+            addMetadataProperty(metadata, "height", height);
+        }
+
+        List<String> extraClasses = new ArrayList<>();
+
+        for (String clazz : attrs.getClasses()) {
+            if (Pattern.matches("^wp-image.*", clazz)) {
+                String attachmentIdString = clazz.replace("wp-image-", "");
+                if (NumberUtils.toInt(attachmentIdString) != 0) {
+                    addMetadataProperty(metadata, "attachment_id", attachmentIdString);
+                } else {
+                    AppLog.d(AppLog.T.EDITOR, "AttachmentId was not an integer! String value: " + attachmentIdString);
+                }
+            } else if (Pattern.matches("^align.*", clazz)) {
+                addMetadataProperty(metadata, "align", clazz.replace("align-", ""));
+            } else if (Pattern.matches("^size-.*", clazz)) {
+                addMetadataProperty(metadata, "size", clazz.replace("size-", ""));
+            } else {
+                extraClasses.add(clazz);
+            }
+        }
+
+        addMetadataProperty(metadata, "classes", TextUtils.join(" ", extraClasses));
+
+        return metadata;
+    }
+
+    private static JSONObject addMetadataProperty(JSONObject jsonObject, String key, String value) {
+        try {
+            return jsonObject.put(key, value);
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+
+        return jsonObject;
+    }
+
+    private static JSONObject addMetadataProperty(JSONObject jsonObject, String key, int value) {
+        try {
+            return jsonObject.put(key, value);
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+
+        return jsonObject;
+    }
+
+    private static JSONObject addMetadataProperty(JSONObject jsonObject, String key, boolean value) {
+        try {
+            return jsonObject.put(key, value);
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+
+        return jsonObject;
+    }
+
+    static Set<String> getClassAttribute(Attributes attributes) {
+        if (attributes.getIndex("class") == -1) {
+            return new HashSet<>(new ArrayList<String>());
+        }
+        return new HashSet<>(Arrays.asList(attributes.getValue("class").split(" ")));
+    }
+
+    static class AttributesWithClass {
+        private AztecAttributes aztecAttributes;
```

Comment:
That's the code style suggestion for the AOSP project, not for Android apps in general. Hungarian notation serves no purpose and I fully agree with JW here: http://jakewharton.com/just-say-no-to-hungarian-notation/. If you insist, though, I'll rename it.